### ISSUE TITLE
[Awaiting C+ Payment confirmation] [HOLD for payment 2023-08-21] [$1000]  Inconsistent file error result on uploading unsupported file type

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#19718]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Open [staging.new.expensify.com](http://staging.new.expensify.com/)
2. Send a .exe file to a chat
3. Send a .tar.gz file to a chat

## Expected Result: 
Both should have consistent unsupported file type error
## Actual Result: 
While trying to attach .exe file returned error dialog immediately, trying to attach .tar.gz file doesn't returned the same dialog error and returned the error on sending the message instead
## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.19-2
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/43996225/82d5f9aa-bcfc-4c93-a62a-8a3969593492

https://github.com/Expensify/App/assets/43996225/71e6e163-7eb1-4023-b0e3-4cbf969fe1a7

**Expensify/Expensify Issue URL:**
**Issue reported by:** @kerupuksambel
**Slack conversation:** 
https://expensify.slack.com/archives/C049HHMV9SM/p1684986305893649
[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~018f83b776223928ef</li>
        <li>Upwork Job ID: 1663615039149555712</li>
        <li>Last Price Increase: 2023-06-06</li>
    </ul>
</details>

**Comment by PrashantMangukiya** [comment]
## Proposal

Posting proposal early as per new [guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#propose-a-solution-for-the-job)

### Please re-state the problem that we are trying to solve in this issue.
Inconsistent file error result on uploading unsupported file type

### What is the root cause of that problem?
Not allowed file type checking done via variable  from const file here
https://github.com/Expensify/App/blob/12d2f3c89de6c0271ae90760de816e198d3e9025/src/CONST.js#L41-L95

We can see `gz` extension not added in this. So it will not show error while file selected with `gz` extension. This is the root cause of the problem.

### What changes do you think we should make in order to solve the problem?
We have to add `gz` extension within `UNALLOWED_EXTENSIONS` array as shown under. It will solve the issue.
```javascript
UNALLOWED_EXTENSIONS: [ 
   ...
  'sys',
  'gz',    // *** Add this
  'vb',
   ...
]
```

### What alternative solutions did you explore? (Optional)
None

**Comment by trjExpensify** [comment]
Yeah, seems clear to me. Though, @Beamanator are we just going to keep running into this maintaining a list of "unallowed filetypes" versus a list of allowed ones and then throwing the front-end error if it's not one of them? 😕

**Comment by Beamanator** [comment]
@trjExpensify gooood question, I think there's another issue somewhere else where someone has been working on allowing all file types? Or many more or something like that? I've really only mainly worked on image file type stuff for avatars sorry :D

**Comment by amyevans** [comment]
I believe it's @youssef-lr

**Comment by youssef-lr** [comment]
I'm not really sure if we should fix this, or, as a few engineers have suggested, remove this limitation and allow all filetypes to be uploaded. We initially implemented this as a measure of security to protect users from downloading potentially harmful files in public rooms. I think I'll bring this up and Slack to see if we really need to be blocking certain filetypes or not.

**Comment by youssef-lr** [comment]
Discussion here: https://expensify.slack.com/archives/C03TQ48KC/p1685473892542269

**Comment by amyevans** [comment]
Thanks!

**Comment by amyevans** [comment]
I bumped the thread to confirm if this is the path we'd like to pursue:

> remove the list of unallowed extensions from both [App](https://github.com/Expensify/App/blob/12d2f3c89de6c0271ae90760de816e198d3e9025/src/CONST.js#L41-L95) and [Web-E](https://github.com/Expensify/Web-Expensify/blob/main/lib/ReportUtils.php#L98)

**Comment by trjExpensify** [comment]
Cool, seems like we have agreement yeah?

**Comment by amyevans** [comment]
Yes, let's field an external proposal to remove the `UNALLOWED_EXTENSIONS` list and any related logic from the FE.

cc @PrashantMangukiya in case you'd like to update your proposal

**Comment by PrashantMangukiya** [comment]
@amyevans Sure, let me update proposal to remove `UNALLOWED_EXTENSIONS` list and any related logic from the FE.

**Comment by youssef-lr** [comment]
@amyevans @PrashantMangukiya I think we want to keep this internal as I'm working on allowing all filetypes in the backend as well.

**Comment by amyevans** [comment]
@youssef-lr couldn't the work be split in 2 (FE external/BE internal)? Or if not, would you like to take over on this GH?

**Comment by trjExpensify** [comment]
Bump @youssef-lr, you want this one?

**Comment by trjExpensify** [comment]
Cool, cool.. assigning you the issue.

**Comment by youssef-lr** [comment]
I'm gonna get to this as soon as I'm done with a high priority SNH issue.

**Comment by trjExpensify** [comment]
👋 with the feature freeze in place @youssef-lr, will you be able to get PRs in review this week before this one breaks WAQ?

**Comment by youssef-lr** [comment]
PRs in review.

**Comment by sobitneupane** [comment]
PR in review.

**Comment by trjExpensify** [comment]
What are the next steps here to square this away @youssef-lr @sobitneupane?

**Comment by youssef-lr** [comment]
Looks like there's an error unrelated to the App PR @sobitneupane, I've identified the issue in the backend and will be working in it as part of this [issue](https://github.com/Expensify/App/issues/21795) because it has the same root cause. I think we should hold on merging that app PR until I have the fix merged.

**Comment by amyevans** [comment]
Still holding, looks like https://github.com/Expensify/Web-Expensify/pull/38169 is on staging but not prod

**Comment by amyevans** [comment]
https://github.com/Expensify/Web-Expensify/pull/38169 has been live for quite a bit now 😅. I'll remove the hold from the title here, @youssef-lr can you fix conflicts in https://github.com/Expensify/App/pull/21285 and remove the hold?

**Comment by youssef-lr** [comment]
Woops, on it.

**Comment by trjExpensify** [comment]
PR hit staging 4 days ago, Melv.

**Comment by sobitneupane** [comment]
> **BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

The cause of the initially reported issue is missing of certain file type in `UNALLOWED_EXTENSIONS`. We got rid of all those `UNALLOWED_EXTENSIONS` and now all file types are allowed.

**Comment by sobitneupane** [comment]
**Regression Test Proposal**

1. Upload files with extension .exe and .tar.gz
2. Verify that those files are uploaded successfully without any error.

Do we agree 👍 or 👎

**Comment by sobitneupane** [comment]
Requested payment on newDot.

**Comment by kerupuksambel** [comment]
> @kerupuksambel - sent you an offer!

Accepted the offer, thank you!

**Comment by trjExpensify** [comment]
Paid! Updating the title to await confirmation of @sobitneupane's.

**Comment by amyevans** [comment]
@trjExpensify Do we need to keep it open for confirmation of C+ payment? I think on other issues I've seen them closed prior.

**Comment by JmillsExpensify** [comment]
Reviewed the details for @sobitneupane. $1,000 approved for payment in NewDot based on BZ summary.
